lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -48,6 +48,12 @@ def setup_rabbitmq_connection_config
         # Time in seconds to wait before retrying a connection
         config :connect_retry_interval, :validate => :number, :default => 1
 
+        # The default connection timeout; zero means wait indefinitely
+        config :connection_timeout, :validate => :number
```

Comment:
If we have a non-nil default value, I agree we'll need a way to signal "no timeout" - and maybe the simplest way is to use 0 or -1 (select(2) uses -1 for no timeout, for example). +1 for using a special number in this case, even if it feels weird, mainly because there's no way to "unset" or nullify a setting in the config.